packages/http/src/validator/index.ts

```diff
@@ -96,21 +96,20 @@ const findResponseByStatus = (responses: IHttpOperationResponse[], statusCode: n
 
 export const validateMediaType = (contents: NonEmptyArray<IMediaTypeContent>, mediaType: string) =>
   pipe(
-    doOption
-      .bind('parsedMediaType', pipe(O.fromNullable(mediaType), O.map(contentType.parse)))
-      .doL(({ parsedMediaType }) =>
-        pipe(
-          contents,
-          findFirst(c => {
-            const parsedSelectedContentMediaType = contentType.parse(c.mediaType);
-            return (
-              !!typeIs(parsedMediaType.type, [parsedSelectedContentMediaType.type]) &&
-              isMatch(parsedMediaType.parameters, parsedSelectedContentMediaType.parameters)
-            );
-          })
-        )
+    O.fromNullable(mediaType),
+    O.map(contentType.parse),
+    O.chain(parsedMediaType =>
+      pipe(
+        contents,
+        findFirst(c => {
```

Comment:
Totally minor, but if we're around then maybe:
```suggestion
        A.findFirst(c => {
```

Reply:
13f478e